samza-core/src/main/java/org/apache/samza/coordinator/stream/CoordinatorStreamSystemConsumer.java

```diff
@@ -290,4 +290,21 @@ public boolean hasNewMessages(SystemStreamPartitionIterator iterator) {
     return iterator.hasNext();
   }
 
+  /**
+   * Has the SystemConsumer started.
+   *
+   * @return True if started, otherwise false.
+   */
+  public boolean isStarted() {
```

Comment:
I don't think isBootstrapped should be a flag. The 'bootstrap' method is a misnomer, it should be called 'refresh' or something. It's meant to be called multiple times, once before each use, to get the latest stream state.

Reply:
Thanks for clarifying. I'll remove that flag.